### PR TITLE
Update Bitbucket docs for workspaces migration

### PR DESCRIPTION
**Pull request by mzalimeni**
# Description
Update "teams" terminology to new "workspaces", and refresh external doc
links.

Also fix some minor typos.

# Reasons
See [Comms Brief](https://docs.google.com/document/d/1PNdWU-KgWIXBsSg6fUF0W3IoPoHZAac5gnN7cw32nDA/edit). These changes are in support of our preparation for the Bitbucket teams/users deprecation.

### CODE REVIEW

**Comment by mzalimeni** [comment]
cc @sarahseaton-circleci

File: jekyll/_cci2/gh-bb-integration.md [hunk 1]
```diff
@@ -202,7 +202,7 @@ This section provides an overview of the possible team and individual account ch
 
 2. If an individual upgrades to a GitHub Team account they can add team members and may give admin permissions on the repo to those who run builds. The owner of the team GitHub account (org) must go to the CircleCI [Add Project](https://circleci.com/add-projects){:rel="nofollow"}, click the link to GitHub's application permissions screen, and select Authorize CircleCI to enable members of the org to follow the project from their account. A team account with two members is $25 per month instead of $7 per month for a personal account.
 
-3. An individual Bitbucket account is free for private repos for teams of up to five. An individual may create a Bitbucket team, add members and give out admin permissions on the repo as needed to those who need to build. This project would appear in CircleCI for members to follow without additional cost.
+3. An individual Bitbucket account is free for private repos for teams of up to five. An individual may create a Bitbucket workspace, add members and give out admin permissions on the repo as needed to those who need to build. This project would appear in CircleCI for members to follow without additional cost.
```
Comment (thomasr8) [inline jekyll/_cci2/gh-bb-integration.md:205]:
```suggestion
3. An individual Bitbucket account is free for private repos for teams of up to five. An individual may create a Bitbucket workspace, add members, and give out admin permissions on the repo as needed to those who need to build. This project would appear in CircleCI for members to follow without additional cost.
```

File: jekyll/_cci2/security-server.adoc [hunk 2]
```diff
@@ -100,5 +100,5 @@ If you are getting started with CircleCI there are some things you can ask your
...
 - Use Restricted Contexts with teams to share environment variables with a select security group. Read through the <<contexts#restricting-a-context,contexts>> document to learn more.
 - Ensure you audit who has access to SSH keys in your organization.
-- Ensure that your team is using Two-Factor Authentication (2FA) with your VCS (https://help.github.com/en/articles/securing-your-account-with-two-factor-authentication-2fa[Github 2FA], https://confluence.atlassian.com/bitbucket/two-step-verification-777023203.html[Bitbucket]). If a user's GitHub or Bitbucket account is compromised a nefarious actor could push code or potentially steal secrets.
+- Ensure that your team is using Two-Factor Authentication (2FA) with your VCS (https://help.github.com/en/articles/securing-your-account-with-two-factor-authentication-2fa[Github 2FA], https://support.atlassian.com/bitbucket-cloud/docs/enable-two-step-verification/[Bitbucket]). If a user's GitHub or Bitbucket account is compromised a nefarious actor could push code or potentially steal secrets.
```
Comment (thomasr8) [inline jekyll/_cci2/security-server.adoc:103]:
```suggestion
- Ensure that your team is using Two-Factor Authentication (2FA) with your VCS (https://help.github.com/en/articles/securing-your-account-with-two-factor-authentication-2fa[Github 2FA], https://support.atlassian.com/bitbucket-cloud/docs/enable-two-step-verification/[Bitbucket]). If a user's GitHub or Bitbucket account is compromised, a nefarious actor could push code or potentially steal secrets.
```

**Review by thomasr8 — APPROVED** [review]
Just a few minor comments, but overall, looks really good 👍

**Review by melyalex — APPROVED** [review]

**Comment by mzalimeni** [comment]
After some late-breaking discoveries regarding the upcoming migration, we may need to alter our approach s.t. these docs remain as they are, with minor updates to links.

Writing up a quick summary of this for sharing in our [#bitbucket-deprecations-2020](https://circleci.slack.com/archives/C01B5UK9724) channel, but for now, closing this PR to avoid doc churn until we have everyone on board.

**Pull request closed without merging**